### DISCUSSION TITLE
How should I set up an alarm for a response time greater than 5 seconds for a specific URI?

### DISCUSSION THREAD

**Discussion by abelzha** [apache/skywalking#11501] [Q&A] [answered]
skywalking version: 7.0.0

**I attempted the following two approaches, but none of them triggered an alert.**

1.oal：endpoint_resp_time_user = from(Endpoint.latency).filter(name == "/user/getList").longAvg();
  user_resp_time_test_rule:
    metrics-name: endpoint_resp_time_user 
    op: ">"
    threshold: 5000
    period: 5
    count: 3
    silence-period: 10
    message: xxxxxxxxxxxx.

2.oal in official_analysis.oal: endpoint_avg = from(Endpoint.latency).longAvg();

  user_resp_time_test_rule:
    metrics-name: endpoint_avg
    op: ">"
    threshold: 5000
    period: 5
    count: 3
    silence-period: 10 
    message:xxxxxxxxxxxxxx.
    include-names:
      - /user/getList

**Comment by wu-sheng** [comment]
7.0 was released over two years ago. read the latest alerting doc.

**Reply by abelzha** [reply]
got it , thx

**Reply by wu-sheng** [reply]
And, the name of endpoint should reference from UI, rather than putting the format randomly.

**Reply by abelzha** [reply]
yes , the endpoint "/user/getList"   i copied from UI

**Reply by wu-sheng** [reply, marked as answer]
Choose the latest, and AFAIK, the latest endpoint format from the latest Java agent doesn't like this.

Anyway, it is pointless we are talking about a 2 years gap. Learn the latest first, and we can continue.

**Reply by abelzha** [reply]
Alright, thank you very much!
